File: v1/src/main/java/com/api/v1/domain/car/Car.java
package com.api.v1.domain.car;

import java.time.ZonedDateTime;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import com.api.v1.dtos.car.CarUpdatingRequestDto;

import lombok.Getter;

@Getter
@Document(collection = "v1_cars")
public class Car {

    @Id
    private UUID id;

    @Field
    private String make;

    @Field
    private String model;

    @Field
    private int productionYear;

    @Field
    private String vin;

    @Field
    private String plateNumber;

    @Field
    private String addedAt;

    @Field
    private String updatedAt;

    @Field
    private Set<String> plateNumbers = new HashSet<>();

    public Car(
            UUID id,
            String make,
            String model,
            int productionYear,
            String vin,
            String plateNumber,
            String addedAt
    ) {
        this.id = id;
        this.make = make;
        this.model = model;
        this.productionYear = productionYear;
        this.vin = vin;
        this.plateNumber = plateNumber;
        this.addedAt = addedAt;
    }

    public void update(CarUpdatingRequestDto requestDto) {
        this.make = requestDto.make();
        this.model = requestDto.model();
        this.productionYear = requestDto.productionYear();
        this.plateNumber = requestDto.plateNumber();
        this.updatedAt = ZonedDateTime.now().toString();
    }

}

File: v2/src/main/kotlin/com/api/v2/car/domain/Car.kt
package com.api.v2.car.domain

import com.api.v2.car.dtos.CarUpdateRequestDto
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.ZonedDateTime
import java.util.UUID

@Document(collection = "v2_cars")
class Car(
    @Id
    val id: UUID,
    val vin: String,
    var model: String,
    var productionYear: String,
    var make: String,
    var plateNumber: String,
    val createdAt: String,
) {

    var updatedAt: String? = null

    fun update(requestDto: CarUpdateRequestDto) {
        model = requestDto.model
        productionYear = requestDto.productionYear
        make = requestDto.make
        plateNumber = requestDto.plateNumber
        updatedAt = ZonedDateTime.now().toString()
    }

}